### ISSUE TITLE
Please provide example of a Flask server push

### ISSUE THREAD

**Issue by buddha314** [miguelgrinberg/Flask-SocketIO#682]
More generally than my question here: https://stackoverflow.com/questions/49673258/javascript-client-will-not-continuously-listen-with-flask-socketio I'm interested in seeing an example where a Flask server pushes events continuously.  For instance, suppose the server is controlling the position of a bouncing ball after the user pushes the "throw ball!" button.  And thank you very much for the library!

**Comment by miguelgrinberg** [comment]
The example in this repository does that: https://github.com/miguelgrinberg/Flask-SocketIO/blob/master/example/app.py#L25

**Comment by buddha314** [comment]
I am trying to augment your code with the bouncing ball.  It looks like the line

```
global thread
    with thread_lock:
        if thread is None:
            print("starting back_work")
            thread = socketio.start_background_task(target=back_work)
```

means you can only have one thread (no duh, Yoda).  What is the proper way to add another thread?  Thanks!

**Comment by buddha314** [comment]
Also, how do I stop a thread?

**Comment by miguelgrinberg** [comment]
Not sure I understand what you are asking. The example I pointed you at uses one thread. If you need more than one, then just create as many as you need.

A thread stops when the top-level function for the thread returns.

**Comment by buddha314** [comment]
Yes, I'm being too brief.

You started a thread called, unironically, `thread`.  I wasn't sure if that was a key word, so in adding a new background task, I also tested `if thread is None`.  Since the first task was running, this failed and my new task did not go off.  This is a result of me being a crappy programmer.  And I work a lot with strongly typed languages like Chapel, so I'm not used to things not being declared as types.

I added a new object called `pony_thread` to keep track of the new background task and that worked fine.

Next, I'm wondering how to push a button and make your function

```
def background_thread():
    """Example of how to send server generated events to clients."""
    count = 0
    while True:
        socketio.sleep(10)
        count += 1
        socketio.emit('my_response',
                      {'data': 'Server generated event', 'count': count}, namespace='/test')
```

stop.  Thanks for your patience!
